### PR TITLE
Added introduction page to the supply reorder form

### PR DESCRIPTION
**Pull request by carlosfelix2**
## Summary
The PR adds the introduction page to the supply reorder form. 
* Note that there are still a few styling and content changes still coming for the page and those will be addressed as they get finalized
* Also note that the rest of the form pages are still not developed, so only review the introduction page
* Also, also note that this PR **WILL NOT** merge to main, but to our integration branch to keep working on the form. To see all changes simply start creating a new PR and compare this branch to main.

See [Figma file](https://www.figma.com/design/o9zkSuKTzHm9eQqHTEN1iz/Medical-supplies-reorder?node-id=2698-24599&node-type=frame&t=aFIc4DS5OK5umZyn-0) - make sure you are in the "Ready for dev" page
## Related issue(s)
Tickets are coming, but not ready yet.
You can see [in this Slack canvas](https://dsva.slack.com/docs/T03FECE8V/F07T0HRP430) the current list of tasks/use cases

## Testing done
* Added unit tests
* E2E tests will come at a later date as the testing framework for forms is different from other apps and requires learning
* Manual testing
  * Add the following lines to your `content-build/src/applications/registry.json` file 
  ``` json
  {
      "appName": "Order health care supplies",
      "entryName": "mhv-supply-reordering",
      "rootUrl": "/my-health/order-supplies",
      "productId": "",
      "template": {
        "vagovprod": true,
        "layout": "page-react.html"
      }
    },
  ```
  * Run the mocks: `yarn mock-api --responses src/applications/mhv-supply-reordering/mocks/index.js`
  * Run the UI: `yarn watch --env entry=mhv-supply-reordering`
  * Go to the page http://localhost:3001/my-health/order-supplies/
  * Change the MDOT response by changing the user in `src/applications/mhv-supply-reordering/mocks/index.js` line 6

### Veteran is eligible, and has both available and unavailable supplies
localhost_3001_my-health_order-supplies_introduction_default

### Veteran is eligible, and only has available supplies
localhost_3001_my-health_order-supplies_introduction (4)_no-unaval

### We have an unknown error like a server error
localhost_3001_my-health_order-supplies_introduction (3)

### Veteran is deceased
localhost_3001_my-health_order-supplies_introduction

### Veteran is an invalid user
localhost_3001_my-health_order-supplies_introduction (1)

### Veteran has no supplies to reorder
localhost_3001_my-health_order-supplies_introduction (2)

### Veteran is not eligible to reorder
localhost_3001_my-health_order-supplies_introduction (5)

### CODE REVIEW

File: src/applications/mhv-supply-reordering/components/IneligibleErrorMessage.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { format } from 'date-fns';
+
+const IneligibleErrorMessage = ({ nextAvailabilityDate }) => {
+  const formattedDate = format(new Date(nextAvailabilityDate), 'MMMM d, yyyy');
+  return (
+    <va-alert status="warning" data-testid="mhv-supply-intro-ineligible">
```
Comment (carlosfelix2) [inline src/applications/mhv-supply-reordering/components/IneligibleErrorMessage.jsx:8]: Note: This error is a copy from the old supply reorder form.

File: src/applications/mhv-supply-reordering/constants.js [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+  'Use this form to order hearing aid batteries and accessories, and CPAP supplies.';
 
 export const MDOT_API_URL = `${environment.API_URL}/v0/in_progress_forms/mdot`;
 export const MDOT_API_STATES = Object.freeze({
   ERROR: 'ERROR',
   SUCCESS: 'SUCCESS',
   PENDING: 'PENDING',
 });
+
+export const dlcServiceTelephone = (
```
Comment (carlosfelix2) [inline src/applications/mhv-supply-reordering/constants.js:15]: Move this to be its own component

File: src/applications/mhv-supply-reordering/utilities/mdot/index.jsx [hunk 3]
```diff
@@ -1,9 +1,16 @@
+import sortBy from 'lodash/sortBy';
```
Comment (carlosfelix2) [inline src/applications/mhv-supply-reordering/utilities/mdot/index.jsx:1]: This should be a JS file. Check on other files too that do not use React

File: src/applications/mhv-supply-reordering/utilities/mdot/index.jsx [hunk 4]
```diff
@@ -37,12 +44,33 @@ export const getAvailableSupplies = mdotData => {
...
+  if (
+    !isEligible(mdotData) &&
+    mdotData?.supplies &&
+    mdotData.supplies.length > 0
+  ) {
+    const sortedByAvailability = sortBy(
+      mdotData.supplies.filter(supply => supply.availableForReorder === true),
+      'nextAvailabilityDate',
+    );
+    return head(sortedByAvailability)?.nextAvailabilityDate || null;
```
Comment (carlosfelix2) [inline src/applications/mhv-supply-reordering/utilities/mdot/index.jsx:69]: Remove the use of `head`.

File: src/applications/mhv-supply-reordering/utilities/mdot/index.jsx [hunk 5]
```diff
@@ -37,12 +44,33 @@ export const getAvailableSupplies = mdotData => {
...
+ * @param {*} mdotData the MDOT data
+ * @returns the next eligibility date if exists, or null if no date
+ */
+export const getEligibilityDate = mdotData => {
+  if (
+    !isEligible(mdotData) &&
+    mdotData?.supplies &&
+    mdotData.supplies.length > 0
+  ) {
+    const sortedByAvailability = sortBy(
```
[the remark this answers is not in the document]
Comment (carlosfelix2) [reply src/applications/mhv-supply-reordering/utilities/mdot/index.jsx]: Done

File: src/applications/mhv-supply-reordering/utilities/mdot/index.jsx (hunk 4 above)
Comment (carlosfelix2) [reply src/applications/mhv-supply-reordering/utilities/mdot/index.jsx]: Done.

File: src/applications/mhv-supply-reordering/utilities/mdot/index.jsx (hunk 3 above)
Comment (carlosfelix2) [reply src/applications/mhv-supply-reordering/utilities/mdot/index.jsx]: Done

File: src/applications/mhv-supply-reordering/constants.js (hunk 2 above)
Comment (carlosfelix2) [reply src/applications/mhv-supply-reordering/constants.js]: Done.

File: src/applications/mhv-supply-reordering/actions/index.js [hunk 6]
```diff
@@ -27,19 +41,19 @@ export const fetchMdotData = () => async dispatch => {
   dispatch(initiateApiCall());
   apiRequest(MDOT_API_URL)
     .then(body => {
-      if (body.errors) {
-        const mdotError = head(body.errors);
+      if (body.errors && body.errors.length > 0) {
```
Comment (dcloud) [inline src/applications/mhv-supply-reordering/actions/index.js:44]: I was thinking object/array destructuring, e.g. `let { errors : [mdotError = null] = [] } = body;` or similar, but what you have is more readable.

File: src/applications/mhv-supply-reordering/components/IneligibleErrorMessage.jsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+import { format } from 'date-fns';
+import DlcServiceTelephone from './DlcSupportTelephone';
+
+/**
+ * Generates an alert for a veteran that is not eligible to order supplies.
+ * @param {string} nextAvailabilityDate the date supplies will be available to reorder
+ * @returns the alert
+ */
+const IneligibleErrorMessage = ({ nextAvailabilityDate }) => {
+  const formattedDate = format(new Date(nextAvailabilityDate), 'MMMM d, yyyy');
```
Comment (dcloud) [inline src/applications/mhv-supply-reordering/components/IneligibleErrorMessage.jsx:12]: Does this need to handle the possibility of the string not being parseable as a date?

File: src/applications/mhv-supply-reordering/containers/App.jsx [hunk 8]
```diff
@@ -32,13 +44,18 @@ App.propTypes = {
   location: PropTypes.shape({
     basename: PropTypes.string,
   }),
+  user: PropTypes.object,
 };
 
+const mapStateToProps = state => ({
+  user: state.user,
```
Comment (dcloud) [inline src/applications/mhv-supply-reordering/containers/App.jsx:51]: Do we know why we'd do this over using `selectUser`?

File: src/applications/mhv-supply-reordering/sass/mhv-supply-reordering.scss [hunk 9]
```diff
@@ -4,3 +4,8 @@
 @import "~@department-of-veterans-affairs/css-library/dist/stylesheets/modules/m-modal";
 @import "~@department-of-veterans-affairs/css-library/dist/stylesheets/modules/m-omb-info";
 @import "../../../platform/forms/sass/m-form-confirmation";
+@import "~@department-of-veterans-affairs/formation/sass/shared-variables";
+
+.mhv-c-reorder-unavail-card {
+    max-width: 486px;
```
Comment (dcloud) [inline src/applications/mhv-supply-reordering/sass/mhv-supply-reordering.scss:10]: I know design is still being worked on, so just noting that this seems like an arbitrary value. Unsure if there are VADS variables that could get the intended effect

**Review by dcloud — APPROVED** [review]
Thanks for going over this yesterday. I think all of the changes look good to move forward. I noted some minor things I noticed, but nothing that should block this from being merged into your larger development branch

File: src/applications/mhv-supply-reordering/containers/App.jsx (hunk 8 above)
Comment (carlosfelix2) [reply src/applications/mhv-supply-reordering/containers/App.jsx:51]: Is there a preference? I thought it was the same.

**Review by nichia — APPROVED** [review]
Thanks for walking thru the PR. It looks good to me

**Review by radavis — APPROVED** [review]
Looks good! It's nice to see lots of unit specs.

**Pull request merged**